public/app/features/explore/state/history.ts

```diff
@@ -79,20 +77,31 @@ export const addHistoryItem = (
   queries: DataQuery[]
 ): ThunkResult<void> => {
   return async (dispatch, getState) => {
-    const { richHistoryStorageFull, limitExceeded } = await addToRichHistory(
-      datasourceUid,
-      datasourceName,
-      queries,
-      false,
-      '',
-      !getState().explore.richHistoryStorageFull,
-      !getState().explore.richHistoryLimitExceededWarningShown
-    );
-    if (richHistoryStorageFull) {
-      dispatch(richHistoryStorageFullAction());
-    }
-    if (limitExceeded) {
-      dispatch(richHistoryLimitExceededAction());
+    // always add to local storage
+    getLocalRichHistoryStorage().addToRichHistory({
```

Comment:
I think it'd be better to wrap it in `try/catch` as it may throw exceptions and prevent the remote history from being saved.